src/backend/distributed/commands/cluster.c

```diff
@@ -69,6 +70,20 @@ PreprocessClusterStmt(Node *node, const char *clusterCommand,
 		return NIL;
 	}
 
+	/* we do not support CLUSTER command on partitioned tables as it can not be run inside transaction blocks */
+	if (PartitionedTableNoLock(relationId))
```

Comment:
maybe we can set `task->cannotBeExecutedInTransction = false;` on the tasks

Reply:
This was a good starting step, but is not enough. Apparently we can not use `worker_apply_shard_ddl_command` to run CLUSTER commands on shards.

I think we should not spend more time to add support for this, and continue with warning messages.

```
NOTICE:  00000: issuing SELECT worker_apply_shard_ddl_command (102045, 'public', 'CLUSTER sale USING sale_pk;')
DETAIL:  on server hanefi@localhost:9702 connectionId: 2
LOCATION:  LogRemoteCommand, remote_commands.c:351
WARNING:  01000: CONCURRENTLY-enabled index commands can fail partially, leaving behind an INVALID index.
 Use DROP INDEX CONCURRENTLY IF EXISTS to remove the invalid index.
LOCATION:  ExecuteDistributedDDLJob, utility_hook.c:1382
ERROR:  25001: CLUSTER cannot be executed from a function
CONTEXT:  while executing command on localhost:9702
LOCATION:  ReportResultError, remote_commands.c:318
```

Reply:
I think we can still do that, we should also avoid `worker_apply_shard_ddl_command` such as we do for `VACUUM` -- `VacuumTaskList()`.

I'm fine with skipping it today given the code-freeze.
 
maybe clarify the comment further:

```
PostgreSQL currently does not support CLUSTER command on partitioned tables in a transaction block. Although Citus can execute commands outside of transaction block -- such as VACUUM -- we cannot do that here because CLUSTER command is also not allowed from a function call as well. By default, Citus uses `worker_apply_shard_ddl_command()`, where we should avoid it for this case.
```